cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -22,29 +22,34 @@
 
 namespace cudf::io::detail::parquet {
 
+namespace {
+
+/**
+ * @brief Recursively copy the output buffer from one to another.
+ *
+ * This only copies `name` and `user_data` fields, which are generated during reader construction.
+ *
+ * @param buff The old output buffer
+ * @param new_buff The new output buffer
+ */
+void copy_output_buffer(column_buffer const& buff, column_buffer& new_buff)
```

Comment:
How about `column_buffer::create_empty_like` to avoid two-step creation process and make the name clearer?

Reply:
This is great! I Implemented that. Thanks.